### PR TITLE
Standardize go package layout

### PR DESCRIPTION
**Pull request by Yancey0623**
This PR standardized go package layout as https://github.com/golang-standards/project-layout .

Major changes:
1. `sql/` => `/pkg/sql/`
1. `server/` => `pkg/server/`
1. `sqlfs` => `pkg/sqlfs/`
1. `sql/python/` => `python`
1. update `doc/walkthrough.md`

### CODE REVIEW

File: scripts/test_hive.sh [hunk 1]
```diff
@@ -34,7 +34,7 @@ export SQLFLOW_TEST_DB=hive
 # NOTE: we have already installed sqlflow_submitter under python installation path
 # using latest develop branch, but when testing on CI, we need to use the code in
 # the current pull request.
-export PYTHONPATH=$GOPATH/src/sqlflow.org/sqlflow/sql/python
```
Comment (tonyyang-svail) [inline scripts/test_hive.sh:37]: why removing `src/`?

File: scripts/test_hive.sh [hunk 2]
```diff
@@ -34,7 +34,7 @@ export SQLFLOW_TEST_DB=hive
...
+export PYTHONPATH=$GOPATH/sqlflow.org/sqlflow/python
```
Comment (tonyyang-svail) [inline scripts/test_hive.sh]: On line 47,

```
python -m unittest discover -v sql/python "db_test.py"
```
=>
```
python -m unittest discover -v python "db_test.py"
```

File: scripts/test_ipython.sh [hunk 3]
```diff
@@ -43,7 +43,7 @@ DATASOURCE="mysql://root:root@tcp(127.0.0.1:3306)/?maxAllowedPacket=0"
 # NOTE: we have already installed sqlflow_submitter under python installation path
 # using latest develop branch, but when testing on CI, we need to use the code in
 # the current pull request.
-export PYTHONPATH=$GOPATH/src/sqlflow.org/sqlflow/sql/python
+export PYTHONPATH=$GOPATH/sqlflow.org/sqlflow/python
```
Comment (tonyyang-svail) [inline scripts/test_ipython.sh]: why removing `src/`?

File: scripts/test_maxcompute.sh [hunk 4]
```diff
@@ -22,7 +22,7 @@ fi
 # NOTE: we have already installed sqlflow_submitter under python installation path
 # using latest develop branch, but when testing on CI, we need to use the code in
 # the current pull request.
-export PYTHONPATH=$GOPATH/src/sqlflow.org/sqlflow/sql/python
+export PYTHONPATH=$GOPATH/sqlflow.org/sqlflow/python
```
Comment (tonyyang-svail) [inline scripts/test_maxcompute.sh]: why removing `src/`?

File: scripts/test_units.sh [hunk 5]
```diff
@@ -21,7 +21,7 @@ export SQLFLOW_TEST_DB=mysql
 # NOTE: we have already installed sqlflow_submitter under python installation path
 # using latest develop branch, but when testing on CI, we need to use the code in
 # the current pull request.
-export PYTHONPATH=$GOPATH/src/sqlflow.org/sqlflow/sql/python
+export PYTHONPATH=$GOPATH/sqlflow.org/sqlflow/python
```
Comment (tonyyang-svail) [inline scripts/test_units.sh]: why removing `src/`?

File: scripts/test_hive.sh (hunk 2 above)
Comment (Yancey0623) [reply scripts/test_hive.sh]: Thx, done.

File: scripts/test_maxcompute.sh (hunk 4 above)
Comment (Yancey0623) [reply scripts/test_maxcompute.sh]: fixed.

File: scripts/test_hive.sh (hunk 1 above)
Comment (Yancey0623) [reply scripts/test_hive.sh:37]: fixed.

**Review by tonyyang-svail — APPROVED** [review]
LGTM

**Pull request merged**